terraform/modules/ecr/iam.tf

```diff
@@ -60,8 +60,8 @@ data "aws_iam_policy_document" "task_assume_role" {
   }
 }
 
-resource "aws_iam_role" "task" {
-  name               = "${var.environment_name}-task"
+resource "aws_iam_role" "webapp_task" {
+  name               = "${var.environment_name}-webapp-task"
```

Comment:
this is more useful I think, even if they do end up with the same policy. 

I was also wondering whether we need `ecs` in the name, given that IAM roles are not explicitly tied to ECS. I don't think the word `task` is overloaded as a technical term, but it's a fairly generic word and I wonder if it would still make things a bit easier to digest if this role was explicit about being for ecs